src/rust/runtime/memory/buffer/demibuffer.rs

```diff
@@ -0,0 +1,820 @@
+// Copyright (c) Microsoft Corporation.
```

Comment:
Please use the Rust's comment style to document the exposed API.

Reply:
Is that documented somewhere?

Reply:
https://rust-lang.github.io/api-guidelines/

Reply:
We decided in today's group meeting to move forward with this PR and have a group discussion later about using rustdoc to document our code and how we want to do that.